### PR TITLE
Fix "Starting metrics collection from sentinel context" errors

### LINKED ISSUE #9048: Regression in log context tracking
On both matrix.org and jki.re I'm seeing (on latest develop) a lot of:

```
2021-01-07 18:39:07,534 - synapse.util.metrics - 113 - WARNING - None - Starting metrics collection from sentinel context: metrics will be lost
```

which may mean a bunch of our metrics are inaccurate

### PR DESCRIPTION
**Pull request by erikjohnston**
Fixes #9048

### CODE REVIEW

File: changelog.d/9053.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Various clean-ups to the structured logging and logging context code.
```
Comment (richvdh) [inline changelog.d/9053.misc:1]: I kinda wonder if this is a different `bugfix`. It should mean that we correctly track some CPU usage that was being lost before.

**Review by richvdh — APPROVED** [review]

File: changelog.d/9053.misc (hunk 1 above)
Comment (erikjohnston) [reply changelog.d/9053.misc:1]: True, though tbh the measure block looks pretty pointless, its not doing very much in there at all

Comment (richvdh) [reply changelog.d/9053.misc:1]: doesn't it go off and wake up all the syncs and things?

Comment (erikjohnston) [reply changelog.d/9053.misc]: It just needs to look up each user and room in the dicts and call `.notify()`. I've updated the changelog anyway

**Pull request merged**